test/GetRequests.cs

```diff
@@ -0,0 +1,125 @@
+using System.Net.Http;
+using NUnit.Framework;
+using Ovh.Api;
+using Ovh.Api.Testing;
+using System;
+using FakeItEasy;
+using System.Linq;
+using Ovh.Test.Models;
+
+namespace Ovh.Test
+{
+    [TestFixture]
+    public class GetRequests
+    {
+        static long currentClientTimestamp = 1566485765;
+        static long currentServerTimestamp = 1566485767;
+        static DateTimeOffset currentDateTime = DateTimeOffset.FromUnixTimeSeconds(currentClientTimestamp);
+        static ITimeProvider timeProvider = A.Fake<ITimeProvider>();
+
+        public GetRequests()
+        {
+            A.CallTo(() => timeProvider.UtcNow).Returns(currentDateTime);
+        }
+
+        public static void MockAuthTimeCallWithFakeItEasy(FakeHttpMessageHandler fake)
+        {
+            A.CallTo(() =>
+                fake.Send(A<HttpRequestMessage>.That.Matches(
+                    r => r.RequestUri.ToString().Contains("/auth/time"))))
+                .Returns(Responses.Get.time_message);
+        }
+
+        [Test]
+        public void GET_auth_time()
+        {
+            var testHandler = A.Fake<FakeHttpMessageHandler>(a => a.CallsBaseMethods());
+            MockAuthTimeCallWithFakeItEasy(testHandler);
+
+            var httpClient = new HttpClient(testHandler);
+            var c = new Client("ovh-eu", httpClient: httpClient).AsTestable(timeProvider);
+
+            Assert.AreEqual(2, c.TimeDelta);
+        }
+
+        [Test]
+        public void GET_me_as_string()
+        {
+            var fake = A.Fake<FakeHttpMessageHandler>(a => a.CallsBaseMethods());
+            MockAuthTimeCallWithFakeItEasy(fake);
+
+            A.CallTo(() =>
+                fake.Send(A<HttpRequestMessage>.That.Matches(
+                    r => r.RequestUri.ToString().Contains("/me"))))
+                .Returns(Responses.Get.me_message);
+
+
+            var c = ClientFactory.GetClient(fake);
+            var result = c.GetAsync("/me").Result;
```

Comment:
Global thought about test methods : it seems that NUnit [supports async methods](https://stackoverflow.com/questions/12191831/how-do-i-test-an-async-method-with-nunit-or-possibly-with-another-framework), would you give it a try?

Reply:
Yup, done. Not sure why I didn't :astonished: